src/pgp.rs

```diff
@@ -62,14 +69,26 @@ impl KeyTrait for SignedPublicKeyOrSubkey<'_> {
             Self::Subkey(k) => k.algorithm(),
         }
     }
-}
 
-impl PublicKeyTrait for SignedPublicKeyOrSubkey<'_> {
+    fn created_at(&self) -> &chrono::DateTime<chrono::Utc> {
+        match self {
```

Comment:
Both `SignedPublicKey` and `SignedPublicSubKey` implement `PulicKeyTrait`, so i think these `match`es everywhere can be replaced with a function returning a reference to the trait, though there are not many of them

Reply:
Got it. Maybe `pgp` should switch to dynamic dispatch then, i.e. change
```Rust
fn serialize_for_hashing(&self, writer: &mut impl Write) -> Result<()>;
```
to
```Rust
fn serialize_for_hashing(&self, writer: &mut dyn Write) -> Result<()>;
```
(and the same for `encrypt()`, though it needs `CryptoRng + Rng`, so it will be more complex) or that would affect performance?